xbrowse/datastore/elasticsearch_datastore.py

```diff
@@ -258,42 +264,99 @@ def get_elasticsearch_variants(
                     variant_id_filter_term |= q_obj
             s = s.filter(variant_id_filter_term)
 
-        if indivs_to_consider:
-            atleast_one_nonref_genotype_filter = None
-            for indiv_id in indivs_to_consider:
+        genotype_filters = {}
+        for key, value in query_json.items():
+            if key.startswith("genotypes"):
+                indiv_id = ".".join(key.split(".")[1:-1])
                 sample_id = family_individual_ids_to_sample_ids.get(indiv_id) or indiv_id
                 encoded_sample_id = _encode_name(sample_id)
-                q = Q('range', **{encoded_sample_id+"_num_alt": {'gte': 1}})
-                if atleast_one_nonref_genotype_filter is None:
-                    atleast_one_nonref_genotype_filter = q
-                else:
-                    atleast_one_nonref_genotype_filter |= q
+                genotype_filter = value
+                if type(genotype_filter) == int or type(genotype_filter) == basestring:
+                    genotype_filters[encoded_sample_id] = [('term', genotype_filter)]
```

Comment:
I don't think you can guarantee in the current system that  encoded_sample_id is unique across families, etc.

Reply:
This is copy-paste: https://github.com/macarthur-lab/seqr/pull/661/files/c45f3809551a30512065c2f56c109574a9b33a24#diff-bb4e5d5e6d2fce2f9ce02a49e3098e78L368

Under the current system a sample_id has to be unique within the callset or nothing works. For new search it might be wiorth revisiting that constraint but this just amintains existing behavior